packages/gen-ai/frontend/src/app/Chatbot/components/ViewCodeModal.tsx

```diff
@@ -0,0 +1,99 @@
+/* eslint-disable camelcase */
+import * as React from 'react';
+import {
+  Modal,
+  ModalVariant,
+  ModalHeader,
+  ModalBody,
+  Spinner,
+  Alert,
+  AlertVariant,
+} from '@patternfly/react-core';
+import { CodeEditor, Language } from '@patternfly/react-code-editor';
+import { exportCode } from '~/app/services/llamaStackService';
+import { CodeExportRequest } from '~/app/types';
+
+interface ViewCodeModalProps {
+  isOpen: boolean;
+  onToggle: () => void;
+  input?: string;
+  model?: string;
+  systemInstruction?: string;
+}
+
+const ViewCodeModal: React.FunctionComponent<ViewCodeModalProps> = ({
+  isOpen,
+  onToggle,
+  input,
+  model,
+  systemInstruction,
+}) => {
+  const [code, setCode] = React.useState<string>('');
+  const [isLoading, setIsLoading] = React.useState<boolean>(false);
+  const [error, setError] = React.useState<string>('');
+
+  const isDisabled = !input || !model;
```

Comment:
I am wondering if this is really needed. We can only open the modal when it's not disabled, right?